tests/migration/migration.go

```diff
@@ -2860,25 +2860,34 @@ var _ = SIGMigrationDescribe("VM Live Migration", decorators.RequiresTwoSchedula
 		})
 	})
 
-	Context(" with a dedicated migration network", Serial, func() {
+	Context("with a dedicated migration network", Serial, func() {
+		var migrationNetworkName string
+
 		BeforeEach(func() {
 			virtClient = kubevirt.Client()
 
-			By("Creating the Network Attachment Definition")
-			nad := libmigration.GenerateMigrationCNINetworkAttachmentDefinition()
-			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Create(context.TODO(), nad, metav1.CreateOptions{})
-			Expect(err).NotTo(HaveOccurred(), "Failed to create the Network Attachment Definition")
+			By("Checking if MigrationNetworkName flag is set")
+			if flags.MigrationNetworkName != "" {
+				migrationNetworkName = flags.MigrationNetworkName
+			} else {
+				By("Creating the Network Attachment Definition")
+				nad := libmigration.GenerateMigrationCNINetworkAttachmentDefinition()
+				migrationNetworkName = nad.Name
+				_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Create(context.TODO(), nad, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Failed to create the Network Attachment Definition")
+			}
 
 			By("Setting it as the migration network in the KubeVirt CR")
-			libmigration.SetDedicatedMigrationNetwork(nad.Name)
+			libmigration.SetDedicatedMigrationNetwork(migrationNetworkName)
 		})
 		AfterEach(func() {
 			By("Clearing the migration network in the KubeVirt CR")
 			libmigration.ClearDedicatedMigrationNetwork()
 
-			By("Deleting the Network Attachment Definition")
-			nad := libmigration.GenerateMigrationCNINetworkAttachmentDefinition()
-			Expect(virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Delete(context.TODO(), nad.Name, metav1.DeleteOptions{})).To(Succeed(), "Failed to delete the Network Attachment Definition")
+			if flags.MigrationNetworkName == "" {
+				By("Deleting the Network Attachment Definition")
+				Expect(virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Delete(context.TODO(), migrationNetworkName, metav1.DeleteOptions{})).To(Succeed(), "Failed to delete the Network Attachment Definition")
+			}
```

Comment:
nit: 
You can save this checking (and AfterEach) by deletion the NAD using DeferCleanup right next to the NAD creation at the BeforeEach, for example:
```
BeforeEach(func(){
    ...
    _, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Create(context.Background(), nad, metav1.CreateOptions{})
    Expect(err).NotTo(HaveOccurred())
    DeferCleanup(func() {
        err: = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Create(context.Background(), nad, metav1.CreateOptions{})
        Expect(err).NotTo(HaveOccurred())
  })
})
```